database_management_bot.py

```diff
@@ -0,0 +1,2235 @@
+
+
+@under_maintenance
+def scan_handler(update: Update, context: CallbackContext) -> None:
+    """
+    The scan_handler function is used to tell the user they can scan the barcode of a package.
+
+    :param update: Update: Pass the incoming update to the handler
+    :param context: CallbackContext: Pass data between callbacks
+    :return: None
+    """
+    user = update.message.from_user
+    logger.info("%s: %s", user.first_name, update.message.text)
+
+    reply_keyboard = [['Відмінити сканування']]
+
+    update.message.reply_text(
+        text='Будь ласка, надсилайте мені фото пакувань, де я можу *чітко* побачити штрихкод '
+             'для перевірки наявності\.',
+        parse_mode='MarkdownV2',
+        reply_markup=ReplyKeyboardMarkup(
+            reply_keyboard,
+            one_time_keyboard=True,
+            resize_keyboard=True,
+            input_field_placeholder='Надішліть фото',
+        ),
+    )
+
+
+def get_db_query_result(barcode) -> bool or None:
+    """
+    The get_db_query_result function takes a barcode as an argument and returns the result of a MongoDB query.
+    If no results are found, it returns None.
+
+    :param barcode: Check if the barcode exists in the database
+    :return: A dictionary with the product information if the barcode exists in the database
+    """
+    try:
+        logger.info("Database quired. Checking availability")
+        query_result = collection.find_one({"code": barcode}, {"_id": 0})
+        if query_result is None:
+            return
+
+        return query_result
+    except Exception as e:
+        logger.error(e)
+        return
+
+
+def format_query(query_result) -> str or None:
+    """
+    The format_query function takes a query result and returns a string
+    representation of the query result. This is used for displaying
+    query results in human-readable format.
+
+    :param query_result: Retrieve the information from the database
+    :return: A string containing the name, active ingredient and description of the drug
+    """
+    try:
+        logger.info("Retrieving info")
+        str_output = f"<b>Назва</b>: {query_result['name']} " \
+                     f"\n<b>Діюча речовина</b>: {query_result['active_ingredient']} " \
+                     f"\n<b>Опис</b>: {query_result['description']} "
+        return str_output
+    except Exception as e:
+        logger.info(e)
+        return
+
+
+def retrieve_query_photo(query_result) -> bytes or None:
+    """
+    The retrieve_query_photo function retrieves the photo from a query result.
+
+    :param query_result: Retrieve the photo from the database
+    :return: The binary data of the photo
+    """
+    try:
+        assert query_result is not None
+
+        if query_result['photo'] == b'':
+            logger.info("Field 'photo' is empty")
+            return
+
+        logger.info("Retrieving photo")
+        img = query_result['photo']
+        return img
+    except AssertionError:
+        return
```

Comment:
Unnecessary exception.
```suggestion
    if query_result is None:
        return
        
    if query_result['photo'] == b'':
        logger.info("Field 'photo' is empty")
        return

    logger.info("Retrieving photo")
    img = query_result['photo']
    return img
```

Reply:
Accepted